ocfweb/account/register.py

```diff
@@ -37,8 +42,14 @@ def request_account(request: HttpRequest) -> Union[HttpResponseRedirect, HttpRes
     status = 'new_request'
 
     existing_accounts = search.users_by_calnet_uid(calnet_uid)
+    eligible_group_signatories = groups_by_student_signat(calnet_uid)
 
-    if existing_accounts and calnet_uid not in TESTER_CALNET_UIDS:
+    existing_group_accounts = {}
+    for group_oid in list(eligible_group_signatories.keys()):
+        if group_by_oid(group_oid) and group_oid not in map(operator.itemgetter(0), TEST_GROUP_ACCOUNTS):
+            existing_group_accounts[group_oid] = eligible_group_signatories.pop(group_oid)
```

Comment:
~can't we just `[group_oid]` normally instead of using `pop`~

oh i get it, we are removing groups with accounts that already exist. this code was rly confusing imo but I'm not sure if there's anything to be done about it